unitTests/curvatureResistance/curvatureResistance.py

```diff
@@ -0,0 +1,318 @@
+
+sys.path.append('../..')
+
+from efficiency import totalLossesFunction
+from ocp import casadiSolver
+from train import Train
+from track import Track
+from utils import postProcessDataFrame
+
+
+class TestCurvatureResistance(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        '''
+        Creates two test tracks that are used in the tests.
+        '''
+
+        cls.track = ("../../tracks/", "00_var_speed_limit_100")
+
+        cls.constantK = 1/300 # [m^-1]
+
+        cls.finalPosition = 3475 # [m]
+
+        cls.speedAtStart = 1 # [m/s]
+
+        cls.speedAtEnd = 1 # [m/s]
+
+        cls.thresholdVelocityError= 1e-3 # dimensionless quantity used in the minimum time test
+
+        cls.thresholdEnergyError = 5e-2 # dimensionless quantity used in the minimum energy test
+
+        try:
+
+            cls.trackNoCurvature = Track(config={'id': cls.track[1]}, pathJSON=cls.track[0])
+
+        except Exception as e:
+
+            raise unittest.SkipTest("Impossible to create track object. Error was: ", e)
+
+        cls.trackConstantCurvature = copy.deepcopy(cls.trackNoCurvature)
+
+        cls.trackConstantCurvature.importCurvatureTuples(tuples=[[0.0, str(1/cls.constantK), str(1/cls.constantK)]])
+
+
+    def specificCurvatureResistanceForce(self, g, rho):
+        '''
+        Computes specific curvature resistance force [m/s^2]
+
+        - param g: constant of gravity [m/s^2]
+        - param rho: dimensionless rotating mass factor
+        '''
+
+        return g*0.5*abs(self.constantK)/((1-30*abs(self.constantK))*rho)*(abs(self.constantK)<=1/300) + \
+                g*0.65*abs(self.constantK)/((1-55*abs(self.constantK))*rho)*(abs(self.constantK)>1/300)
+
+
+    def solveOptimalControlProblem(self, track, energyOptimal, lossFunction,
+                                   terminalTime, train=None, finalPosition=None, realLossFunction=None):
+        '''
+        Solves a minimum time or minimium energy optimal control problem.
+
+        - param pathToTrackData: absolute path to the json file containing the track dataset
+        - param trackDataFilename: name of json file without extension
+        - param energyOptimal: bool variable. If false the minimum time problem is solved
+        - param lossFunction: a function taking two inputs (f,v) and returning the electrical
+        losses of the train. Used for solving the minimum energy problem.
+        - param terminalTime: time at which the trip must complete. For the minimum time problem this is just an upperbound of the problem.
+        - param finalPosition: position where the trip must end.
+        - param realLossFunction: if provided this loss function is used to compute the real losses of the train. Note that it is not used to
+        solve the minimum energy problem.
+        - return: a dataframe containing the solution of the problem.
+        '''
+
+        v0 = self.speedAtStart
+        vN = self.speedAtEnd
+
+        if (train == None):
```

Comment:
I would suggest to pass the train "ready" to this function instead the None check.